### ISSUE TITLE
Are multi-nucleotide and complex variants ignored?

### ISSUE THREAD

**Issue by rebber** [bioinform/somaticseq#129]
Hi,

We use somaticseq to just merge variants from Mutect2 and HMF Tools SAGE (the latter as "arbitrary" vcf's), the classification module is not used currently. However we were missing some multi-nucleotide variants (MNVs) in the somaticseq output, so I looked into the somaticseq code for how they are handled. I found that it seems any variants in input vcf's with both REF and ALT with length >1 base are ignored.

I see the following division into SNVs or indels, both in  modify_ssMuTect2.py and splitVcf.py (for preparation of arbitrary vcf's):
```python
if len(vcf_i.refbase) == 1 and len(vcf_i.altbase) == 1:
    snv_out.write( new_line + '\n' )
elif len(vcf_i.refbase) == 1 or len(vcf_i.altbase) == 1:
    indel_out.write( new_line + '\n' )
```
And any other variants, i.e `len(vcf_i.refbase) > 1 and len(vcf_i.altbase) > 1`, will be skipped.

Is it a correct observation that MNVs and complex variants are ignored? What was the reasoning behind setting it up like this? Is there any way to go around it?

We do not want to miss these types of variants, and have to look into other tools if we can't avoid this behaivour with somaticseq.

Best regards
Rebecka

**Comment by litaifang** [comment]
Yeah MNV and complex variants are limitations because they just haven't been our focus. 
For VarDict, MNVs are parsed into multiple SNVs.
I can do similar things for MuTect2 and other outputs, i.e., parse MNVs into multiple SNVs and complex variants into SNVs and indels. If you can give me examples of complex variants and MNV's from those outputs, I can incorporate them.

**Comment by rebber** [comment]
Thanks for a quick reply! 
Primarily we want to keep any MNVs and complex variants together, in order to get proper annotation of them by VEP. We will therefore look into some other solution for variant merging.

**Comment by karlkashofer** [comment]
Hi! Just stumbled over this, we also "lost" a variant. This variant, found by Mutect2, gets removed by somaticseq:

> chr12   25245350        .       C       A,G,T   .       multiallelic    AS_FilterStatus=SITE|SITE|SITE;AS_SB_TABLE=368,557|17,32|7,17|5,8;DP=1071;ECNT=3;GERMQ=93;MBQ=40,44,40,40;MFRL=137,143,119,139;MMQ=60,60,60,60;MPOS=24,17,28;POPAF=7.30,7.30,7.30;TLOD=125.65,55.40,25.05  GT:AD:AF:DP:F1R2:F2R1:SB 0/1/2/3:925,49,24,13:0.049,0.025,0.014:1011:503,25,16,5:422,24,8,8:368,557,29,57
which is bad as it are true somatic mutations, see attached image.

<img width="464" height="1382" alt="Image" src="https://github.com/user-attachments/assets/240eaf07-a04f-45b5-895e-cc868dbeb5e7" />

How can we avoid this ? Any chance you find time to fix it?

**Comment by litaifang** [comment]
Is it acceptable to split a MNV into multiple SNVs, i.e,. turning it into C>A, C>G, and C>T individually (3 lines in the vcf fie output)?

**Comment by karlkashofer** [comment]
better than losing it i guess ! I wonder if i could convince Mutect2 to do that...
currently trying "vt decompose -s" before somaticseq, but in any case a clean solution would be needed.

I very much appreciate your speedy reply !

**Comment by karlkashofer** [comment]
Hmm, splitting it into three single variants does not help, they still do not show up in the resulting Consensus.sSNV.vcf
Do you see any other reason for that ?

> chr12   25245350        .       C       A       .       multiallelic       AS_FilterStatus=SITE|SITE|SITE;AS_SB_TABLE=368,557|17,32|7,17|5,8;DP=1071;ECNT=3;GERMQ=93;MBQ=40,44;MFRL=137,143;MMQ=60,60;MPOS=24;POPAF=7.3;TLOD=125.65;OLD_MULTIALLELIC=chr12:25245350:C/A/G/T        GT:AD:AF:DP:F1R2:F2R1:SB        0/1/./.:925,49:0.049:1011:503,25:422,24:368,557,29,57
chr12   25245350        .       C       G       .       multiallelic       AS_FilterStatus=SITE|SITE|SITE;AS_SB_TABLE=368,557|17,32|7,17|5,8;DP=1071;ECNT=3;GERMQ=93;MBQ=40,40;MFRL=137,119;MMQ=60,60;MPOS=17;POPAF=7.3;TLOD=55.4;OLD_MULTIALLELIC=chr12:25245350:C/A/G/T  GT:AD:AF:DP:F1R2:F2R1:SB        0/./1/.:925,24:0.025:1011:503,16:422,8:368,557,29,57
chr12   25245350        .       C       T       .       multiallelic       AS_FilterStatus=SITE|SITE|SITE;AS_SB_TABLE=368,557|17,32|7,17|5,8;DP=1071;ECNT=3;GERMQ=93;MBQ=40,40;MFRL=137,139;MMQ=60,60;MPOS=28;POPAF=7.3;TLOD=25.05;OLD_MULTIALLELIC=chr12:25245350:C/A/G/T GT:AD:AF:DP:F1R2:F2R1:SB        0/././1:925,13:0.014:1011:503,5:422,8:368,557,29,57

**Comment by karlkashofer** [comment]
OK, found the problem:
FilterMutectCalls had --max-alt-allele-count 2 set => it labelled the variant as "multiallelic" in the FILTER column.
Replacing "multiallelic" with "PASS" made SomaticSeq retain the variable and split it into 3 single allele lines:

> Consensus.sSNV.vcf:chr12        25245350        .       C       A       0.0000  LowQual MDK=1,0,0;NUM_TOOLS=1;AF=0.0496 GT:DP4:CD4:refMQ:altMQ:refBQ:altBQ:refNM:altNM:fetSB:fetCD:zMQ:zBQ:MQ0:VAF      0/1:372,566,17,32:928,10,48,1:59.984:59.8776:46.8454:44.6122:0.208955:1:0.55:0.43:-0.215905:-0.584999:0:0.0496
Consensus.sSNV.vcf:chr12        25245350        .       C       G       0.0000  LowQual MDK=1,0,0;NUM_TOOLS=1;AF=0.026  GT:DP4:CD4:refMQ:altMQ:refBQ:altBQ:refNM:altNM:fetSB:fetCD:zMQ:zBQ:MQ0:VAF      0/1:372,566,7,18:928,10,22,3:59.984:60:46.8454:50.92:0.208955:1.2:0.30:0.00:0.0182146:1.0284:0:0.026
Consensus.sSNV.vcf:chr12        25245350        .       C       T       0.0000  LowQual MDK=1,0,0;NUM_TOOLS=1;AF=0.0137 GT:DP4:CD4:refMQ:altMQ:refBQ:altBQ:refNM:altNM:fetSB:fetCD:zMQ:zBQ:MQ0:VAF      0/1:372,566,5,8:928,10,13,0:59.984:60:46.8454:43.6923:0.208955:1.15385:1.00:1.00:0.0132173:-0.729493:0:0.0137

So, the mistake was on my end. I simply filtered this very unusual call. Switching to --max-alt-allele-count 3 fixes this.
Sorry for the noise, all is good, cheers, K
